pkg/http/handler/timeout.go

```diff
@@ -54,25 +54,25 @@ type timeoutHandler struct {
 // https://golang.org/pkg/net/http/#Handler.
 //
 // The implementation is largely inspired by http.TimeoutHandler.
-func NewTimeoutHandler(h http.Handler, msg string, firstByteTimeout time.Duration, idleTimeout time.Duration, maxDurationTimeout time.Duration) http.Handler {
+func NewTimeoutHandler(h http.Handler, msg string, timeout time.Duration, firstByteTimeout time.Duration, idleTimeout time.Duration) http.Handler {
 	return &timeoutHandler{
-		handler:            h,
-		body:               msg,
-		firstByteTimeout:   firstByteTimeout,
-		idleTimeout:        idleTimeout,
-		maxDurationTimeout: maxDurationTimeout,
-		clock:              clock.RealClock{},
+		handler:          h,
+		body:             msg,
+		timeout:          timeout,
+		firstByteTimeout: firstByteTimeout,
+		idleTimeout:      idleTimeout,
+		clock:            clock.RealClock{},
 	}
 }
 
 func (h *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	ctx, cancel := context.WithCancel(r.Context())
 	defer cancel()
 
-	firstByteTimeout := getTimer(h.clock, h.firstByteTimeout)
-	var firstByteTimeoutDrained bool
+	timeout := getTimer(h.clock, h.timeout)
```

Comment:
It looks like this will also cause a switharoo from current behavior regarding how you can configure these timeouts to be indefinite. Currently, you can't disable TTFB by setting it to zero. With this change, you would be able to disable `firstByteTimeout`, but won't be able to disable the overall timeout.

Can we make it so the user can opt out of both by setting them to 0? (...I'm mostly concerned about websockets / long-polling, FWIW.)

Reply:
I think disabling the overall timeout was a side effect of the incorrect behaviour

@dprotaso I don't know if we want to enable that as an option?

Reply:
IMO it makes to add the option to disable (possibly through another route to avoid perpetuating the incorrect behavior), but I'd defer to @dprotaso on whether that's permissible here